aiida_mlip/data/model.py

```diff
@@ -0,0 +1,159 @@
+    param arch_path: Path to the folder in which the file is stored (str or pathlib.Path).
+    param file: Path to the file (str or pathlib.Path)
+    param architecture: mlip architecture of the model
+    param cache_path: Path to the parent folder of arch_path (str or pathlib.Path)
+    return: A tuple of a PAth and a str containing the path of the file of interest and the architecture
+    """
+    file_hash = _calculate_hash(file)
+    for existing_file in (
+        ex_file
+        for ex_file in arch_path.rglob("*")
+        if ex_file.is_file() and not ex_file.samefile(file)
+    ):
+        if _calculate_hash(existing_file) == file_hash:
+            ex_file_path = Path(existing_file).parent
+            if arch_path == ex_file_path:
+                file.unlink()
+                return existing_file, architecture
+
+            if arch_path == cache_path and arch_path != ex_file_path:
+                architecture = ex_file_path.name
+                file.unlink()
+                return Path(existing_file), architecture
+    return Path(file), architecture
+
+
+class ModelData(SinglefileData):
+    """Class to save a model file as an AiiDA data type.
+    The file can be an existing one or a file to download
+    """
+
+    def __init__(
+        self, file: str | Path, filename: str = None, architecture: str = None, **kwargs
+    ):
+        """Construct a new instance and set the contents to that of the file.
+
+        :param file: absolute path to the file
+        :param filename: specify filename to use (defaults to name of provided file).
+        :param architecture: specify architecture
+        """
+
+        super().__init__(file, filename, **kwargs)
+        self.base.attributes.set("architecture", architecture)
+
+    def set_file(
+        self, file: str | Path, filename: str = None, architecture: str = None, **kwargs
+    ):
+        """Add a file to the node, parse it and set the attributes found.
+
+        :param file: absolute path to the file (str or pathlib.Path)
+        :param filename: specify filename to use (defaults to name of provided file)
+        :param architecture: specify architecture in which the model is used (default = None)
+        """
+
+        super().set_file(file, filename, **kwargs)
+
+        self.base.attributes.set("architecture", architecture)
+
+    @classmethod
+    def local_file(
+        cls, file: str | Path, filename: str = None, architecture: str = None
+    ):
+        """Save a local file as ModelData
+        :param file: path to the file (str or pathlib.Path)
+        :param filename: specify filename to use (defaults to name of provided file)
+        :param architecture: specify architecture in which the model is used (default = None)
+        """
+        file_path = Path(file).resolve()
+        return cls(file=file_path, filename=filename, architecture=architecture)
+
+    @classmethod
+    # pylint: disable=R0913
```

Comment:
`too-many-arguments` is probably preferable to the code, so it's clear what you're disabling.

It might also be worth adding a value for `max-args = ` under `[tool.pylint.format]` in pyproject.toml if this is needed more often. I prefer more having the option of more than five generally, as long as it's something to still consider.